woocommerce/woocommerce-gutenberg-products-block
# Add extensibility point for extensions to filter payment methods

This PR introduces a new way for extensions to register their own callbacks to determine wheater a payment method is supported.
Fixes #4621 
[Tests](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/4622), [documentation](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/4623), and the [Typescript conversion](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/4687) will be added in follow-up tickets.

### How to test the changes in this Pull Request:
1. Make sure Check payments and Cash on delivery payment methods are activated and appear in the Checkout block
2. Import the registration function somewhere in an extension
```js
import { registerPaymentMethodExtensionCallbacks} from '@woocommerce/blocks-registry';
```
3. Add some callbacks 
```js
registerPaymentMethodExtensionCallbacks( 'woocommerce-marketplace-extension',
	{
		cod: ( arg ) => { return false; },
                 // this zipcode works only for Germany
		cheque: ( arg ) => { return arg.billingData.postcode === "10473";}
	}
);
```
4. Go to the Checkout block
3. Notice that Cash on delivery is no longer available
4. Notice that Check payments are available only when the Postcode for Billing data is 10473



### Changelog

> Add extensibility point for extensions to filter payment methods.
